src/core/Akka.TestKit/TestKitBase_AwaitConditions.cs

```diff
@@ -80,31 +67,12 @@ public async Task AwaitConditionAsync(Func<bool> conditionIsFulfilled)
         /// specified in config value "akka.test.timefactor".</param>
         public void AwaitCondition(Func<bool> conditionIsFulfilled, TimeSpan? max)
         {
-            var maxDur = RemainingOrDilated(max);
-            var interval = new TimeSpan(maxDur.Ticks / 10);
-            var logger = _testState.TestKitSettings.LogTestKitCalls ? _testState.Log : null;
-            InternalAwaitCondition(conditionIsFulfilled, maxDur, interval, (format, args) => _assertions.Fail(format, args), logger);
+            AwaitConditionAsync(conditionIsFulfilled, max).AsTask()
+                .WaitAndUnwrapException();
         }
         
-        /// <summary>
-        /// <para>Await until the given condition evaluates to <c>true</c> or the timeout
-        /// expires, whichever comes first.</para>
-        /// <para>If no timeout is given, take it from the innermost enclosing `within`
-        /// block (if inside a `within` block) or the value specified in config value "akka.test.single-expect-default". 
-        /// The value is <see cref="Dilated(TimeSpan)">dilated</see>, i.e. scaled by the factor 
-        /// specified in config value "akka.test.timefactor"..</para>
-        /// <para>A call to <paramref name="conditionIsFulfilled"/> is done immediately, then the threads sleep
-        /// for about a tenth of the timeout value, before it checks the condition again. This is repeated until
-        /// timeout or the condition evaluates to <c>true</c>. To specify another interval, use the overload
-        /// <see cref="AwaitCondition(System.Func{bool},System.Nullable{System.TimeSpan},System.Nullable{System.TimeSpan},string)"/>
-        /// </para>
-        /// </summary>
-        /// <param name="conditionIsFulfilled">The condition that must be fulfilled within the duration.</param>
-        /// <param name="max">The maximum duration. If undefined, uses the remaining time 
-        /// (if inside a `within` block) or the value specified in config value "akka.test.single-expect-default". 
-        /// The value is <see cref="Dilated(TimeSpan)">dilated</see>, i.e. scaled by the factor 
-        /// specified in config value "akka.test.timefactor".</param>
-        public async Task AwaitConditionAsync(Func<bool> conditionIsFulfilled, TimeSpan? max)
+        /// <inheritdoc cref="AwaitCondition(Func{bool}, TimeSpan?)"/>
+        public async ValueTask AwaitConditionAsync(Func<bool> conditionIsFulfilled, TimeSpan? max)
```

Comment:
Need to add CancellationToken support. Can't change old implementation, have to stick to Task instead of ValueTask